jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -228,9 +228,56 @@ YUI.add('user-profile', function() {
     switchEnv: function(uuid, name) {
       var props = this.props;
       props.showConnectingMask();
+      this.close();
       props.switchModel(uuid, this.state.envList);
       props.dbEnvironmentSet('name', name);
+    },
+
+    /**
+      Set to a new, blank canvas with no model.
+
+      @method resetModel
+    */
+    resetModel: function() {
```

Comment:
The logic here and the logic in `switchEnv` above are nearly identical and should probably be abstracted out and moved into the apps `switchEnv` passed through the `switchModel`